src/transformStateWithClones.js

```diff
@@ -65,7 +65,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const transrormsHistory = [];
+  const copyState = { ...state };
+
+  for (const transform of transforms) {
+    if (transform.operation === 'addProperties') {
```

Comment:
I think it's better to use switch case for code's readability, but no strong opinion ;)